### PR TITLE
scheduelrs: add load expectations for hot scheudler

### PR DESCRIPTION
**Pull request by nolouch**
Signed-off-by: nolouch <nolouch@gmail.com>

### What problem does this PR solve? 
To prevent excessive redundant scheduling
image        
### What is changed and how it works?
add the avg load as an expectation.

### Check List 

Tests 
 - Unit test
 - Manual test
the scheduler operator is reduced and QPS is stable.
image

### CODE REVIEW

File: server/schedulers/hot_region_config.go [hunk 1]
```diff
@@ -42,6 +42,7 @@ func initHotRegionScheduleConfig() *hotRegionSchedulerConfig {
 		GreatDecRatio:         0.95,
 		MinorDecRatio:         0.99,
 		MaxPeerNum:            1000,
+		ToleranceRatio:        1.02,
```
Comment (rleungx) [inline server/schedulers/hot_region_config.go]: Why 1.02?

File: pkg/mock/mockcluster/mockcluster.go [hunk 2]
```diff
@@ -420,10 +426,13 @@ func (mc *Cluster) UpdateStorageReadBytes(storeID uint64, bytesRead uint64) {
 }
 
 // UpdateStorageWrittenKeys updates store written keys.
-func (mc *Cluster) UpdateStorageWrittenKeys(storeID uint64, keysWritten uint64) {
+func (mc *Cluster) UpdateStorageWrittenKeys(storeID uint64, keysWritten uint64, updateSingleOption ...struct{}) {
```
Comment (rleungx) [inline pkg/mock/mockcluster/mockcluster.go]: How about:
```suggestion
func (mc *Cluster) UpdateStorageWrittenStats(storeID uint64, keysWritten uint64, bytesWritten uint64) {
```

File: server/schedulers/hot_region.go [hunk 3]
```diff
@@ -721,9 +763,8 @@ func (bs *balanceSolver) calcProgressiveRank() {
 	rank := int64(0)
 	if bs.rwTy == write && bs.opTy == transferLeader {
 		// In this condition, CPU usage is the matter.
-		// Only consider about count and key rate.
-		if srcLd.Count > dstLd.Count &&
-			srcLd.KeyRate >= dstLd.KeyRate+peer.GetKeyRate() {
+		// Only consider about key rate.
```
Comment (lhy1024) [inline server/schedulers/hot_region.go]: What is its effect

File: server/schedulers/hot_region_config.go (hunk 1 above)
Comment (nolouch) [reply server/schedulers/hot_region_config.go]: 2%

File: server/schedulers/hot_region.go (hunk 3 above)
Comment (nolouch) [reply server/schedulers/hot_region.go]: The `srcCount` may higher but the `keyRate` lower.

**Pull request merged**